includes/amp-helper-functions.php

```diff
@@ -270,6 +270,53 @@ function amp_register_default_scripts( $wp_scripts ) {
 	}
 }
 
+/**
+ * Generate HTML for AMP scripts that have not yet been printed.
+ *
+ * This is adapted from `wp_scripts()->do_items()`, but it runs only the bare minimum required to output
+ * the missing scripts, without allowing other filters to apply which may cause an invalid AMP response.
+ * The HTML for the scripts is returned instead of being printed.
+ *
+ * @since 0.7.2
+ * @see WP_Scripts::do_items()
+ * @see AMP_Base_Embed_Handler::get_scripts()
+ * @see AMP_Base_Sanitizer::get_scripts()
+ *
+ * @param array $scripts Script handles mapped to URLs or true.
+ * @return string HTML for scripts tags that have not yet been done.
+ */
+function amp_render_scripts( $scripts ) {
```

Comment:
It's nice how this is decomposed into `amp_render_scripts()`, to allow using it for Legacy Templating and theme support.